Catappult/appcoins-wallet-android
# [APPC-2963] The wallet provided does not have enough funds

**What does this PR do?**

   Resolves the bug with "The wallet provided does not have enough funds".

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] AppcoinsRewardsBuyPresenter.kt
- [ ] PaymentMethodsAnalytics.kt
- [ ] PaymentMethodsFragment.kt
- [ ] PaymentMethodsPresenter.kt

**How should this be manually tested?**

Purchase
1. Have a positive balance
2. Buy with test driver app several times without opening the wallet.
3. When the balance will become insufficient the appcoins payment method should not be accessible
 
Top up
1. Have insufficient or zero balance
2. Make sure the appcoins payment method is not be accessible in the test driver app
3. Top up in the wallet app but don’t close the success screen
4. Make sure the appcoins payment method became accessible in the test driver app

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-2963](https://aptoide.atlassian.net/browse/APPC-2963)

**Code Review Checklist**
